### PR TITLE
Hrcpp 451/update apidoc

### PR DESCRIPTION
**Pull request by rigazilla**
Improved doxygen documentation.

https://issues.jboss.org/browse/HRCPP-451

### CODE REVIEW

**Comment by rigazilla** [comment]
@alanfx just added more comments for doxygen. and a useless class removed

File: include/infinispan/hotrod/ConfigurationBuilder.h [hunk 1]
```diff
@@ -21,10 +21,22 @@ using namespace infinispan::hotrod::event;
 namespace infinispan {
 namespace hotrod {
 
+/**
+ * ClusterConfigurationBuilder is used to configure the clusters that are part of the system.
+ * Is a builder of ClusterConfiguration objects.
```
Comment (alanfx) [inline include/infinispan/hotrod/ConfigurationBuilder.h]: "ClusterConfigurationBuilder is used to configure the clusters that are part of the system and is a builder of ClusterConfiguration objects."

File: include/infinispan/hotrod/ConfigurationBuilder.h [hunk 2]
```diff
@@ -36,26 +48,46 @@ class ClusterConfigurationBuilder
...
- * Configuration classes for near cache
+ * Configuration classes for near cache subsystem
  */
 class NearCacheConfigurationBuilder : public ConfigurationChildBuilder
 {
 public:
     NearCacheConfigurationBuilder(ConfigurationBuilder& _builder) : ConfigurationChildBuilder(_builder) {}
 
+    /**
+     * \return the max number of entries that the near cache can contains
```
Comment (alanfx) [inline include/infinispan/hotrod/ConfigurationBuilder.h]: contains-> contain

File: include/infinispan/hotrod/ConfigurationBuilder.h [hunk 3]
```diff
@@ -36,26 +48,46 @@ class ClusterConfigurationBuilder
...
+     */
     int getMaxEntries() const {
         return m_maxEntries;
     }
 
+    /**
+     * Set the max number of entries that the near cache can contains
```
Comment (alanfx) [inline include/infinispan/hotrod/ConfigurationBuilder.h]: contains-> contain

File: include/infinispan/hotrod/ConfigurationBuilder.h [hunk 4]
```diff
@@ -21,10 +21,22 @@ using namespace infinispan::hotrod::event;
...
 {
 public:
 	ClusterConfigurationBuilder(std::vector<ServerConfigurationBuilder>& servers, ConfigurationBuilder &parent) : servers(servers), m_parent(parent) {}
+	/**
+	 * Add a server node to this cluster
+	 *
+	 * \param host hostname
+	 * \param port tcp port number
+	 *
+	 * \return this object for further configuration
```
Comment (alanfx) [inline include/infinispan/hotrod/ConfigurationBuilder.h]: You used "return this object for fluent configuration" below, and I think that is a better message

File: include/infinispan/hotrod/ConfigurationBuilder.h [hunk 5]
```diff
@@ -99,6 +130,13 @@ class ConfigurationBuilder
 
      void validate() {}
 
+    /**
+     *  Add a new cluster to the configuration builder
+     *
+     * \param clusterName an name identifier for the new cluster
```
Comment (alanfx) [inline include/infinispan/hotrod/ConfigurationBuilder.h]: an name identifier for the new cluster->the name of the new cluster

File: include/infinispan/hotrod/NearCacheConfiguration.h [hunk 6]
```diff
@@ -14,7 +14,20 @@
 namespace infinispan {
 namespace hotrod {
 
-enum NearCacheMode { DISABLED=0, INVALIDATED=1 };
+/**
+ * Enumeration of all the near cache mode of operation
+ */
+enum NearCacheMode {
+    DISABLED=0,     /*!< No near cache. */
+    INVALIDATED=1   /*!< Near cache in invalidated mode. In invalidated mode an entry is create near only after a get
```
Comment (alanfx) [inline include/infinispan/hotrod/NearCacheConfiguration.h]: In invalidated mode an entry is create near-> In invalidated mode an entry in the near cache is created

File: include/infinispan/hotrod/Query.h [hunk 7]
```diff
@@ -21,14 +21,36 @@ namespace client {
...
+/**
+ * This class represent a query request.
+ */
 class QueryRequest : public QueryRequestProtobuf
 {
 public:
   inline bool has_jpqlstring() const { return has_querystring(); }
   inline void clear_jpqlstring() { clear_querystring(); }
+  /**
+   * Use querystring() instead
```
Comment (alanfx) [inline include/infinispan/hotrod/Query.h]: I know Deprecated is used by Java, but do you want to use something like that for these comments?

File: include/infinispan/hotrod/RemoteCache.h [hunk 8]
```diff
@@ -252,8 +253,8 @@ template<class K, class V> class RemoteCache: private RemoteCacheBase
      * \param key  key with which the specified value is to be associated
      * \param val  value value to be associated with the specified key
      * \param lifespan  lifespan of the entry.  Negative values are interpreted as unlimited lifespan.
-     *  \param lifespanUnit  unit of measurement for the lifespan
-     *  \param maxIdle  the maximum amount of time this key is allowed to be idle for before it is considered as
+     * \param lifespanUnit  unit of measurement for the lifespan
```
Comment (alanfx) [inline include/infinispan/hotrod/RemoteCache.h]: Should this refer to `TimeUnit`, so users know which units are valid?

I think this is true of all of the `TimeUnit` parameters

File: include/infinispan/hotrod/RemoteCache.h [hunk 9]
```diff
@@ -267,7 +268,11 @@ template<class K, class V> class RemoteCache: private RemoteCacheBase
 
     /**
      * Asynchronous version of put()
-     *    See the synchronous doc for parameters not explained here
+     * \param key  key with which the specified value is to be associated
+     * \param val  value value to be associated with the specified key
+     * \param lifespan  lifespan of the entry.  Negative values are interpreted as unlimited lifespan.
+     * \param maxIdle  the maximum amount of time this key is allowed to be idle for before it is considered as
```
Comment (alanfx) [inline include/infinispan/hotrod/RemoteCache.h]: the maximum amount of time this key is allowed to be idle for before-> the maximum amount of time this key is allowed to be idle before

You can use this for all of the `maxIdle` parameters

**Comment by rigazilla** [comment]
@alanfx changes applied. thanks!

**Comment by alanfx** [comment]
@rigazilla LGTM. Merging...

**Pull request merged**